src/index.ts

```diff
@@ -20,72 +20,135 @@
  */
 
 // The only import we need from the Node AuthN library is the Session class.
-import { Session, InMemoryStorage } from "@inrupt/solid-client-authn-node";
+import {
+  Session,
+  InMemoryStorage,
+  ILoginInputOptions,
+} from "@inrupt/solid-client-authn-node";
+import express from "express";
 
-const argv = require("yargs/yargs")(process.argv.slice(2))
-  .describe(
-    "oidcIssuer",
-    "The identity provider at which the user should authenticate."
-  )
-  .alias("issuer", "oidcIssuer")
-  .describe("clientName", "The name of the bootstrapped app.")
-  .demandOption(["oidcIssuer"])
-  .locale("en")
-  .help().argv;
+import {
+  promptClientName,
+  promptIdp,
+  promptPort,
+  promptRegistration,
+  promptStaticClientInfo,
+} from "./prompts";
 
-import express from "express";
-const app = express();
-const PORT = 3001;
-const iriBase = `http://localhost:${PORT}`;
-const storage = new InMemoryStorage();
+type InputOptions = {
+  solidIdentityProvider?: string;
+  clientName?: string;
+  registrationType?: "static" | "dynamic";
+};
+
+type ValidatedOptions = {
+  solidIdentityProvider: string;
+  registrationType: "static" | "dynamic";
```

Comment:
I'd just reorder to reflect the same order as `InputOptions` (i.e. move this line down one line).